packages/ui/src/components/common/TransactionModal/SwapModal.tsx

```diff
@@ -204,6 +206,23 @@ const InnerSwapModal: FC<InnerSwapModalProps> = ({
   waitingForSteps,
   isLoadingTransaction
 }) => {
+  const [isApprovalPlusSwap, setIsApprovalPlusSwap] = useState(false)
+
+  const firstStep = quote?.steps?.[0]
+
+  useEffect(() => {
```

Comment:
Could this just be a simple variable:
const isApprovalPlus = firstStep?.id === 'approve' && firstStep?.items?.[0]?.status === 'incomplete'